tsinfer/inference.py

```diff
@@ -890,23 +980,77 @@ def __init__(
             [inference_site_position, [sample_data.sequence_length]]
         )
         self.position_map[0] = 0
+        self.recombination = np.zeros(self.num_sites)  # TODO: reduce len by 1
+        self.mismatch = np.zeros(self.num_sites)
+
+        # Set the canonical 'recombination' and 'mismatch' arrays to appropriate values
+        # NB 'recombination' must be of length num_sites-1, mismatch of length num_sites
```

Comment:
I think we should simplify this a bit - you can *either* use the low-level ``recombination`` and ``mismatch`` parameters OR the high-level ``recombination_rate`` and ``mismatch_ratio`` but not mix them. Checking this first and deciding which path we're on will simplify things.